### ISSUE TITLE
testing on own dataset

### ISSUE THREAD

**Issue by Puneesh6280** [souvikmajumder26/Land-Cover-Semantic-Segmentation-PyTorch#4]
I tried adding an own tif file to test folder, but showing me an error.
Is there a way to add and test on our own dataset?

**Comment by souvikmajumder26** [comment]
Did you try to test the model with the sample dataset mentioned in the project on your own tif file? In that case the error might be due to the image bands/channels mismatch. If you build a model solely on your own dataset while tweaking the code (if required) as per the tif file bands/channels it should be able to run with a similar test file.

**Comment by AdityaPrakash-Evigway** [comment]
how to inference the code without providing the masks. I wanted to test the model on my dataset which is not labeled.

> Did you try to test the model with the sample dataset mentioned in the project on your own tif file? In that case the error might be due to the image bands/channels mismatch. If you build a model solely on your own dataset while tweaking the code (if required) as per the tif file bands/channels it should be able to run with a similar test file.

**Comment by andrewp-CWF** [comment]
You can comment out lines 98-106 and line 194 in src/test.py to predict on your own images without a ground truth mask.

> how to inference the code without providing the masks. I wanted to test the model on my dataset which is not labeled.
> 
> > Did you try to test the model with the sample dataset mentioned in the project on your own tif file? In that case the error might be due to the image bands/channels mismatch. If you build a model solely on your own dataset while tweaking the code (if required) as per the tif file bands/channels it should be able to run with a similar test file.

**Comment by souvikmajumder26** [comment]
Thanks @andrewp-CWF . That's exactly what transforms the test script (with ground truth masks) into inference script (without ground truth masks). I've added a new inference.py script with the changes that can be run directly without requiring ground truth masks.

Also made a few updates, new configurable parameters for reducing learning rate on plateau have been added.

Closing this issue since it has been resolved.
